src/main/java/Duke.java

```diff
@@ -1,10 +1,102 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    public static void printError(String errMsg){
+        String printContent = "    ____________________________________________________________\n"
+        + "     " + "ERROR: " + errMsg + "!\n"
+        + "    ____________________________________________________________\n";
+        System.out.println(printContent);
+    }
+
     public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        String enterGreet = "    ____________________________________________________________\n"
+                           + "     Hello! I'm Duke\n"
+                           + "     What can I do for you?\n"
+                           + "    ____________________________________________________________\n";
+        String exitPrompt = "    ____________________________________________________________\n"
+                           + "     Bye. Hope to see you again soon!\n"
+                           + "    ____________________________________________________________\n";
+        System.out.println(enterGreet);
+        TodoList todoList = new TodoList();
+        while(true){
+            String line = in.nextLine();
+            System.out.println(line);
+            if(line.equals("bye")){
+                // quit
+                System.out.println(exitPrompt);
+                break;
+            }else{
+                // parse input
+                int funcIdx = line.indexOf(" ");
+                if(funcIdx == -1){
+                    funcIdx = line.length();
+                }
+                String func = line.substring(0, funcIdx);
```

Comment:
perhaps you could give this variable a better name, as it is stored as a `String` but is named `func`

Reply:
I agree with @lihka1202 . I'd suggest to give a variable name like **commandIndex** or **inputIndex** as it it a bit more clear for readers to understand what you are referring to.

Reply:
Thank you for your good suggestions! I have made corrections.